tiledb/sm/serialization/query.h

```diff
@@ -57,8 +57,25 @@ struct QueryBufferCopyState {
       : offset_size(0)
       , data_size(0) {
   }
+
+  /** Copy constructor. */
+  QueryBufferCopyState(const QueryBufferCopyState& rhs)
+      : offset_size(rhs.offset_size)
+      , data_size(rhs.data_size) {
+  }
+
+  /** Assignment operator. */
+  QueryBufferCopyState& operator=(const QueryBufferCopyState& rhs) {
```

Comment:
Might be wise to add the rest of the rule-of-5 functions as `deleted` here.

Reply:
Done